presto-main/src/main/java/com/facebook/presto/metadata/SqlScalarFunction.java

```diff
@@ -112,13 +112,29 @@ protected SqlScalarFunction(
                 literalParameters);
     }
 
+    protected SqlScalarFunction(Signature signature)
+    {
+        this.signature = requireNonNull(signature, "signature is null");
+        checkArgument(signature.getKind() == SCALAR, "function kind must be SCALAR");
+    }
+
     @Override
     public final Signature getSignature()
     {
         return signature;
     }
 
-    public abstract ScalarFunctionImplementation specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry);
+    public abstract ScalarFunctionImplementation specialize(Map<String, Type> types, List<TypeSignature> parameterTypes, TypeManager typeManager, FunctionRegistry functionRegistry);
```

Comment:
fixed in https://github.com/Teradata/presto/commit/d14695078cedda4323a238150100ce38a9e59ed8